src/importexport/musicxml/internal/musicxml/exportxml.cpp

```diff
@@ -2845,10 +2845,15 @@ static void writeAccidental(XmlWriter& xml, const String& tagName, const Acciden
 
 static void wavyLineStart(const Trill* tr, const int number, Notations& notations, Ornaments& ornaments, XmlWriter& xml)
 {
-    // mscore only supports wavy-line with trill-mark
     notations.tag(xml, tr);
     ornaments.tag(xml);
-    xml.tagRaw(u"trill-mark" + color2xml(tr));
+    if (tr->trillType() == TrillType::TRILL_LINE) {
+        xml.tagRaw(u"trill-mark" + color2xml(tr));
+    } else if (tr->trillType() == TrillType::UPPRALL_LINE) {
+        xml.tagRaw(u"other-ornament smufl=\"ornamentBottomLeftConcaveStroke\"" + color2xml(tr));
+    } else if (tr->trillType() == TrillType::DOWNPRALL_LINE) {
+        xml.tagRaw(u"other-ornament smufl=\"ornamentLeftVerticalStroke\"" + color2xml(tr));
+    }
```

Comment:
What if neither of the `if`s above is true? Wouldn't that cause invalid XML to be written?

It may also be nice to see all of them in action in a new unit test.

Reply:
No, the xml would always be valid, because we only add a symbol to the start of the `wavy-line`.
But it looks like not all cases are covered now, so I'll change it into a `switch` statement to clear things up.

Reply:
Ah, I see, I confused "`tagRaw`" with "Writing a start tag" and assumed that there would be "writing an end tag" a few lines below and feared that in the not-covered cases only an end tag would be written.

Anyway, a `switch` definitely seems a good idea here.

Reply:
done